src/main/java/uk/co/real_logic/benchmarks/rtt/Configuration.java

```diff
@@ -0,0 +1,560 @@
+        }
+
+        /**
+         * Create a new instance of the {@link Configuration} class from this builder.
+         *
+         * @return a {@link Configuration} instance
+         */
+        public Configuration build()
+        {
+            return new Configuration(this);
+        }
+    }
+
+    /**
+     * Create a {@link Configuration} instance based on the provided system properties.
+     *
+     * @return a {@link Configuration} instance.
+     */
+    public static Configuration fromSystemProperties()
+    {
+        final Builder builder = new Builder();
+        if (isPropertyProvided(WARM_UP_ITERATIONS_PROP_NAME))
+        {
+            builder.warmUpIterations(intProperty(WARM_UP_ITERATIONS_PROP_NAME));
+        }
+
+        if (isPropertyProvided(WARM_UP_MESSAGES_PROP_NAME))
+        {
+            builder.warmUpNumberOfMessages(intProperty(WARM_UP_MESSAGES_PROP_NAME));
+        }
+
+        if (isPropertyProvided(ITERATIONS_PROP_NAME))
+        {
+            builder.iterations(intProperty(ITERATIONS_PROP_NAME));
+        }
+
+        if (isPropertyProvided(BATCH_SIZE_PROP_NAME))
+        {
+            builder.batchSize(intProperty(BATCH_SIZE_PROP_NAME));
+        }
+
+        if (isPropertyProvided(MESSAGE_LENGTH_PROP_NAME))
+        {
+            builder.messageLength(intProperty(MESSAGE_LENGTH_PROP_NAME));
+        }
+
+        if (isPropertyProvided(SENDER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.senderIdleStrategy(idleStrategyProperty(SENDER_IDLE_STRATEGY_PROP_NAME));
+        }
+
+        if (isPropertyProvided(RECEIVER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.receiverIdleStrategy(idleStrategyProperty(RECEIVER_IDLE_STRATEGY_PROP_NAME));
+        }
+
+        builder.numberOfMessages(intProperty(MESSAGES_PROP_NAME));
+
+        builder.messageTransceiverClass(classProperty(MESSAGE_TRANSCEIVER_PROP_NAME, MessageTransceiver.class));
+
+        return builder.build();
+    }
+
+    private static int checkMinValue(final int value, final int minValue, final String prefix)
+    {
+        if (value < minValue)
+        {
+            throw new IllegalArgumentException(prefix + " cannot be less than " + minValue + ", got: " + value);
+        }
+        return value;
+    }
+
+    private static Class<? extends MessageTransceiver> validateMessageTransceiverClass(
+        final Class<? extends MessageTransceiver> klass)
+    {
+        requireNonNull(klass, "MessageTransceiver class cannot be null");
+        if (isAbstract(klass.getModifiers()))
+        {
+            throw new IllegalArgumentException("MessageTransceiver class must be a concrete class");
+        }
+        try
+        {
+            final Constructor<? extends MessageTransceiver> constructor = klass.getConstructor(MessageRecorder.class);
+            if (isPublic(constructor.getModifiers()))
+            {
+                return klass;
+            }
+        }
+        catch (final NoSuchMethodException e)
```

Comment:
Name exceptions `ignore` if not dealing with them to eliminate the warning.

Reply:
Done